celery_progress/static/celery_progress/celery_progress.js

```diff
@@ -31,37 +53,49 @@ var CeleryProgressBar = (function () {
         }
     }
 
-    async function updateProgress (progressUrl, options) {
-        options = options || {};
-        var progressBarId = options.progressBarId || 'progress-bar';
-        var progressBarMessage = options.progressBarMessageId || 'progress-bar-message';
-        var progressBarElement = options.progressBarElement || document.getElementById(progressBarId);
-        var progressBarMessageElement = options.progressBarMessageElement || document.getElementById(progressBarMessage);
-        var onProgress = options.onProgress || onProgressDefault;
-        var onSuccess = options.onSuccess || onSuccessDefault;
-        var onError = options.onError || onErrorDefault;
-        var onTaskError = options.onTaskError || onError;
-        var onNetworkError = options.onNetworkError || onError;
-        var onHttpError = options.onHttpError || onError;
-        var onDataError = options.onDataError || onError;
-        var pollInterval = options.pollInterval || 500;
-        var resultElementId = options.resultElementId || 'celery-result';
-        var resultElement = options.resultElement || document.getElementById(resultElementId);
-        var onResult = options.onResult || onResultDefault;
-
+    getMessageDetails(result) {
+        if (this.resultElement) {
+            return ''
+        } else {
+            return result || '';
+        }
+    }
 
-        const getMessageDetails = function (result) {
-            if (resultElement) {
-                return ''
+    /**
+     * Process update message data.
+     * @return true if the task is complete, false if it's not, undefined if `data` is invalid
+     */
+    onData(data) {
+        let done = false;
+        if (data.progress) {
+            this.onProgress(this.progressBarElement, this.progressBarMessageElement, data.progress);
+        }
+        if (data.complete === true) {
+            done = true;
+            if (data.success === true) {
+                this.onSuccess(this.progressBarElement, this.progressBarMessageElement, this.getMessageDetails(data.result));
+            } else if (data.success === false) {
+                this.onTaskError(this.progressBarElement, this.progressBarMessageElement, this.getMessageDetails(data.result));
             } else {
-                return result || '';
+                done = undefined;
+                this.onDataError(this.progressBarElement, this.progressBarMessageElement, "Data Error");
+            }
+            if (data.hasOwnProperty('result')) {
+                this.onResult(this.resultElement, data.result);
             }
-        };
+        } else if (data.complete === undefined) {
+            done = undefined;
+            this.onDataError(this.progressBarElement, this.progressBarMessageElement, "Data Error");
+        }
+        return done;
+    }
+
+    async connect() {
```

Comment:
Looks good! I recommend documenting this in the ReadMe so new users won't be confused on whether or not they should be using backwards compatible methods.

Reply:
I'm not sure it's necessary to change the documentation. I think `CeleryProgressBar.initProgressBar` should still be the preferred method because I can't think of a use case where the user would want to create and manage a `CeleryProgressBar` object directly.